cset-workflow/lib/python/jinja_utils.py

```diff
@@ -0,0 +1,119 @@
+# © Crown copyright, Met Office (2022-2024) and CSET contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Useful functions for the workflow."""
+
+import itertools
+from builtins import max, min, zip
+from glob import glob
+
+# Reexport functions for use within workflow.
+__all__ = [
+    "get_models",
+    "restructure_field_list",
+    "sanitise_task_name",
+    # Reexported functions.
+    "max",
+    "min",
+    "zip",
+    "glob",
+]
+
+
+def get_models(rose_variables: dict):
+    """Load per-model configuration into a single object.
+
+    Returns a list of dictionaries, each one containing a per-model
+    configuration.
+    """
+    models = []
+    for model in range(1, 11):
```

Comment:
I assume we can pick up Stephen' s suggestion later around a script that allows users to replicate the model section as many times as they want. For now I am content that order 10 models should be sufficient for most cases.